lib/core/test/unit/Cardano/Wallet/DBSpec.hs

```diff
@@ -186,10 +194,14 @@ once_ xs = void . once xs
 -- | Shorthand for the readTxHistory result type.
 type TxHistory = [(Hash "Tx", (Tx, TxMeta))]
 
--- | Apply the default sort order (descending on time, then by TxId) to a
--- 'TxHistory'.
-sortTxHistory :: TxHistory -> TxHistory
-sortTxHistory = sortBySlot . sortByTxId
+-- | Apply optional filters on slotId and sort using the default sort order
+-- (first time/slotId, then by TxId) to a 'TxHistory'.
+filterTxHistory :: SortOrder -> Range SlotId -> TxHistory -> TxHistory
+filterTxHistory order range =
+    filter (isWithinRange range . slotId . snd . snd)
+    . (if order == Ascending then reverse else id)
```

Comment:
I reckon we ought to use a case statement here, similar to:

```hs
(case order of
    Ascending -> reverse
    Descending -> id)
```
With a case statement that checks all cases, if someone adds a new constructor later on (however unlikely), then the compiler will remind us to consider the new case at this location.

This is philosophically similar to: https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#avoid-wildcard-when-pattern-matching-on-adts

Reply:
👍 Was afraid of the need for `()` and multiline, but it works totally fine.